### PR TITLE
Identifying PII Sdtypes in Metadata

### LINKED ISSUE #1683: Identifying PII Sdtypes in Metadata
### Problem

The SDV offers a variety of sdtypes including:
* statistical sdtypes: numerical, categorical, datetime, boolean
* other sdtypes (for PII values): see [example here](https://docs.sdv.dev/sdv/reference/metadata-spec/sdtypes#additional-sdtypes-domain-specific-concepts-and-pii)

Currently, our metadata auto-detection only identifies statistical sdtypes and many of the other sdtypes (such as PII values) are left as "unknown". We should auto-detect these types in order to make these process easier.

### Solution

Any method of sdtype auto-detection should include checks for PII sdtypes. The algorithm for now can be simple:

1. Take the original column name and strip out any non-alphabet characters. Make everything lowercase 
    a. Eg. Column name "FIRST_NAME" would be converted to "firstname"
2. Match the resulting string against our reference strings (see below). Look for:
    a. Exact matches or
    b. Partial substring matches. For example: "userfirstname" could match with the reference string "firstname"
3. If there is a match, then mark the sdtype associated with the string
   a. Additionally, make sure the metadata marks "pii=True" for the column

### Reference Strings
For now, we should consider sdtypes that are offered explicitly for SDV customers or have been frequently seen.

For more information: [see docs](https://docs.sdv.dev/sdv/reference/metadata-spec/sdtypes#additional-sdtypes-domain-specific-concepts-and-pii)

| Reference String                        | Associated Sdtype      |
|----------------------------------------|------------------------|
| phonenumber                            | phone_number           |
| email                                  | email                  |
| ssn                                    | ssn                    |
| firstname                              | first_name             |
| lastname                               | last_name              |
| countrycode                            | country_code           |
| administrativeunit, state, province    | administrative_unit    |
| stateabbr                              | state_abbr             |
| city                                   | city                   |
| postcode, postalcode, zipcode           | postcode               |
| streetaddress, line1                    | street_address         |
| secondaryaddress, line2                  | secondary_address      |
| latitude                               | latitude               |
| longitude                              | longitude              |
| ipv4, ipv4address                      | ipv4_address           |
| ipv6, ipv6address, ipaddress           | ipv6_address           |
| macaddress                             | mac_address            |
| useragent, useragentstring              | user_agent_string      |
| iban                                   | iban                   |
| swift11, swift                         | swift11                |
| swift8                                 | swift8                 |
| creditcardnumber                       | credit_card_number     |
| vin                                    | vin                    |
| licenseplate, license                   | license_plate          |

### Test Cases

Here are few test cases we can use to check the logic:

* Original column name "USER PHONE NUMBER" should detect sdtype "phone_number"
* Original column name "addr_line_1" should detect sdtype "street_address"
* Original column name: "First Name" should detect sdtype "first_name"
* Original column name: "guest_email" should detect sdtype "email"

### PR DESCRIPTION
**Pull request by R-Palazzo**
CU-86ayr5vuu
Resolve #1683

For info, if there are 2 references in the column name, like `first_name_and_last_name`
it will be the first sdtype in order in `REFERENCE_TO_SDTYPE` that will be assigned. In this case, it will be `first_name`

### CODE REVIEW

**Comment by sdv-team** [comment]
Task linked: [CU-86ayr5vuu SDV - Identifying PII Sdtypes in Metadata #1683](https://app.clickup.com/t/86ayr5vuu)

File: sdv/metadata/single_table.py [hunk 1]
```diff
@@ -250,6 +290,19 @@ def to_dict(self):
...
+        """Detect PII columns.
+
+        Args:
+            column_name (str):
+                The column name to be analyzed.
+        """
+        cleaned_name = re.sub(r'[^a-zA-Z0-9]', '', column_name).lower()
+        return next((
+            sdtype for reference, sdtype in self._REFERENCE_TO_SDTYPE.items()
+            if cleaned_name == reference or reference in cleaned_name
```
Comment (amontanez24) [inline sdv/metadata/single_table.py:303]: if reference == cleaned_name is true, so is reference in cleaned_name so I think you only need the second check

File: sdv/metadata/single_table.py [hunk 2]
```diff
@@ -322,31 +375,34 @@ def _detect_columns(self, data):
...
+                    else:
+                        sdtype = 'unknown'
 
             column_dict = {'sdtype': sdtype}
 
-            if sdtype == 'unknown':
+            if sdtype in self._REFERENCE_TO_SDTYPE.values():
+                column_dict['pii'] = True
+            elif sdtype == 'unknown':
                 column_dict['pii'] = True
```
Comment (amontanez24) [inline sdv/metadata/single_table.py:406]: we can combine these into one if with an or right?

Comment (R-Palazzo) [reply sdv/metadata/single_table.py]: Yes, done in [81ff339](https://github.com/sdv-dev/SDV/pull/1694/commits/81ff339639c3c85c15c7ba40b9d09eca087a456b)

**Review by pvk-developer — APPROVED** [review]
👍🏻 This is great! LGTM!

**Review by amontanez24 — APPROVED** [review]
LGTM!

**Pull request merged**